hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1304,7 +1312,15 @@ func reconcileHostedControlPlane(hcp *hyperv1.HostedControlPlane, hcluster *hype
 	// always reconcile the release image (facilitates rolling forward)
 	hcp.Spec.ReleaseImage = hcluster.Spec.Release.Image
 
-	hcp.Spec.Configuration = hcluster.Spec.Configuration.DeepCopy()
+	if hcluster.Spec.Configuration != nil {
+		hcp.Spec.Configuration = hcluster.Spec.Configuration.DeepCopy()
+		// for compatibility with previous versions of the CPO, the hcp configuration should be
+		// populated with individual fields *AND* the previous raw extension resources.
```

Comment:
That will fail the check that only one of the two is set further down

Reply:
We are setting both on the HostedControlPlane. We are only checking the HostedCluster.